cloudinit/config/schema.py

```diff
@@ -1429,7 +1430,7 @@ def load_doc(requested_modules: list) -> str:
 
 
 def get_schema_dir() -> str:
-    return os.path.join(os.path.dirname(os.path.abspath(__file__)), "schemas")
+    return os.path.join(Path(__file__).parent.parent.parent, "schemas")
```

Comment:
This unfortunately won't work. Only the `cloudinit` directory gets deployed as a unit, not the parent `cloud-init` directory. In order to accommodate this, we'll need to update `setup.py` to write the schema files to correct location on installation.

I would think somewhere like `/usr/lib/cloud-init/schemas/` would be most appropriate, but I don't think we have a definite directory for something like this yes.

@holmanb , thoughts?

Reply:
> This unfortunately won't work. Only the cloudinit directory gets deployed as a unit, not the parent cloud-init directory. In order to accommodate this, we'll need to update setup.py to write the schema files to correct location on installation.

+1, agreed

> I would think somewhere like /usr/lib/cloud-init/schemas/ would be most appropriate, but I don't think we have a definite directory for something like this yes.

Agreed, although /usr/lib is for [libraries, shared objects, and internal executables](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch04s06.html). Maybe a cloud-init directory under [/usr/share](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch04s11.html) would be more appropriate?

Reply:
+1 on `/usr/share/cloud-init/schemas` as that aligns well with FHS for architecture independent data. We have some good examples of json schema files from other packages:
- vscode: /usr/share/code/resources
- /usr/share/lintian/data
- /usr/share/gnome-shell/extensions


Also note when we move forward on this approach we'll may need a one-liner followup PR against [schemastore updating the github URLs for the base schema](https://github.com/SchemaStore/schemastore/pull/2252/files#diff-1b8f038f5afb1158263a1fc83b9c0ca5a7438cceb90adad99d87168f70edc815R785) so that cloud-init schema can be auto-ingested by tools that automatically validate our user-data schema

Reply:
I can see in setup.py there is an existing /share/doc/cloud-init folder in which the schemas folder can also be created . Should it be maintained I should stick with moving the files to /usr/share/cloud-init/schemas .

Reply:
> I can see in setup.py there is an existing /usr/share/doc/cloud-init folder in which the schemas folder can also be created . Should it be maintained I should stick with moving the files to /usr/share/cloud-init/schemas .

Yes, I think that we agree that it should go there.